### PR TITLE
Shortened some which.fish completions.

### PR DESCRIPTION
**Pull request by amiyzku**
## Description

Shortened some which.fish completions.

Related to https://github.com/fish-shell/fish-shell/issues/6981.

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

File: share/completions/which.fish [hunk 1]
```diff
@@ -4,9 +4,9 @@ if which -v >/dev/null 2>/dev/null # GNU
     complete -c which -l skip-alias -d "Ignore option '--read-alias'"
     complete -c which -l read-functions -d "Read shell function definitions from stdin, reporting matching ones on stdout"
     complete -c which -l skip-functions -d "Ignore option '--read-functions'"
-    complete -c which -l skip-dot -d "Skip directories in PATH that start with a dot"
-    complete -c which -l skip-tilde -d "Skip directories in PATH that start with a tilde and executables which reside in the HOME directory"
-    complete -c which -l show-dot -d "If a directory in PATH starts with a dot and a matching executable was found for that path, then print './programname'"
+    complete -c which -l skip-dot -d "Skip directories in PATH starting with a dot"
+    complete -c which -l skip-tilde -d "Skip directories in PATH starting with a tilde and executables in the HOME directory"
```
Comment (mqudsi) [inline share/completions/which.fish:8]: Can shorten this further to "starting with tilde and executables in \$HOME"

File: share/completions/which.fish [hunk 2]
```diff
@@ -4,9 +4,9 @@ if which -v >/dev/null 2>/dev/null # GNU
...
+    complete -c which -l show-dot -d "If a directory in PATH starts with a dot and a matching executable was found, print './programname'"
```
Comment (mqudsi) [inline share/completions/which.fish:9]: "For matches in PATH dirs that start with a dot, print './programname'" is shorter

**Review by mqudsi — COMMENTED** [review]
This is fine, but I made some suggestions to shorten it further. Also you can change "directories" to "dirs" everywhere *if you like*.

**Comment by amiyzku** [comment]
Thank you for the review! I've incorporated your feedback and made the suggested changes. Please take another look and let me know if there's anything else that needs to be addressed.

**Review by mqudsi — CHANGES_REQUESTED** [review]
FYI I had intentionally written `\$HOME` instead of `$HOME` because in a double quoted (`"`) string, `$HOME` will expand to the actual value of `$HOME`. You need to either escape the `$` sign like `\$` or use `'` single quotes instead of double quotes `"`.

**Comment by amiyzku** [comment]
Thank you for the review! I've updated the code to escape  as suggested.

**Review by mqudsi — APPROVED** [review]

**Pull request merged**